Add dedent pkg

ghost opened the pull request:
[![Analytics](https://kubernetes-site.appspot.com/UA-36037335-10/GitHub/.github/PULL_REQUEST_TEMPLATE.md?pixel)]()

PR for adding dedent package to the kubernetes. 
Used to test latest godeps documentation. Next PR will use dedent package in kubectl's describe command. 

cc @janetkuo 
cc @thockin 
See  #17609

Commit:
Add dedent pkg

mhrgoog commented:
@k8s-bot e2e test this issue: #25812

thockin commented:
as discussed you want 2 commits in the same PR.  one for the dep, the other for the consumer.

Commit:
Use dedent for the kubectl commands

The one side effect is that for the "kubectl help" commands a newline
is prepended to output, which will alter the yaml output.

Here we use dedent to format the code to match the output.

hack/update-generated-docs.sh has been run and the affected files have
been added.

Note: for describe.go we added a period to the end of an output message.

mhrgoog commented:
This pull request now has the two commits as discussed.

mhrgoog commented:
@k8s-bot unit test this issue: #IGNORE

We are kick starting this again as it failed before since go4.org has an expired TLS

thockin commented on pkg/kubectl/cmd/describe.go:
```diff
@@ -39,36 +40,39 @@ type DescribeOptions struct {
-	describe_long = `Show details of a specific resource or group of resources.
+var (
+	describe_long = dedent.Dedent(`
+	Show details of a specific resource or group of resources.
```
if we are nit-picking I would indent this one extra level.  But it's minor and not worth re-piushing unless there are other things to change

mhrgoog replied:
done. I had to rebase anyway.

thockin commented:
so the real fun is all of the other kubectl per-command files that follow the same pattern :)  This was just one of them.

pwittrock commented:
@mhrgoog Looks good.  Another rebase is required.  Probably another update to GoDeps.  Rebases will continue to be a problem as long as this is in the queue, so we may want to bump it to the front once all of the critical stuff for 1.3 is in.  I am not sure how we are planning on implementing the feature freeze for 1.3

janetkuo commented:
Is this PR planning to cover all commands? Some commands are missing, e.g. `kubectl rollout`.

janetkuo commented:
We should document the use of dedent package, at least here https://github.com/kubernetes/kubernetes/blob/master/docs/devel/kubectl-conventions.md#command-implementation-conventions

janetkuo commented:
The 2nd and 3rd commits can be squashed into one.

mhrgoog commented:
I have one more commit coming that should have all the commands (including rollout and config and others) using dedent. 

Also I will squash commit two and three and the new one all into one commit.

I will update the documentation in one final commit.

Let me know if that does not sound ok.

Commit:
Document usage of dedent for kubectl commands

janetkuo commented on docs/devel/kubectl-conventions.md:
```diff
@@ -279,15 +279,17 @@ type MineConfig struct {
-  $ %[1]s first
+  mineExample = dedent.Dedent(`
+	  # Run my command's first action
+	  $ %[1]s first
```
Can you remove the `$` sign also?

Also `%[1]s first` looks mysterious to me. Can you change it to something like `kubectl <command> <first_action>`

mhrgoog replied:
Done. Thanks.

janetkuo commented:
:+1: LGTM now

pwittrock commented:
@k8s-bot unit test this issue: #25363 

[artifacts](https://console.cloud.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/25922/kubernetes-pull-test-unit-integration/27769/)

pwittrock commented:
lgtm

mhrgoog commented:
Adding lgtm as I just reran hack/save-godep.sh
